### ISSUE TITLE
Plan Counting Possible Error

### ISSUE THREAD

**Issue by stack72** [hashicorp/terraform#2590]
I just upgraded to terraform 0.6.0

When running:

```
terraform plan -refresh=true -input=false
```

I got the following output

```
Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

~ aws_autoscaling_group.kibana_autoscale_group
    health_check_type: "EC2" => "ELB"

~ aws_instance.dash_instance
    source_dest_check: "" => "1"

~ aws_instance.docker_registry
    source_dest_check: "" => "1"

~ aws_instance.jenkins_master_instance
    source_dest_check: "" => "1"

~ aws_instance.jenkins_slave_instance
    source_dest_check: "" => "1"

~ aws_instance.mumui_instance
    source_dest_check: "" => "1"


Plan: 0 to add, 131 to change, 0 to destroy.
```

131 doesn't seem right since the list only has 6 changes in it

**Comment by phinze** [comment]
cc @radeksimko sounds like we need to revisit the plan counting

**Comment by mitchellh** [comment]
Ah, I noticed this too actually and forgot to open a bug prior to release. For my anecdote: it also seemed to be the "to change" line for me as well.

**Comment by radeksimko** [comment]
I have spotted the same problem - I can confirm it's happening, although not in all cases. I remember seeing this in non-trivial stacks w/ `terraform_remote_state`.

**Comment by pikeas** [comment]
Sorry about the dupe! See the linked issue for a simpler repro - adding a tag to an `aws_instance` results in `terraform plan` counting all my resources under `to change`.

**Comment by radeksimko** [comment]
@pikeas It would help if you could share a relevant part of your TF code & `tfstate` (minus any secrets), but if you can't, that :ok_hand: , I think we can find a way how to reproduce it.

**Comment by jedineeper** [comment]
Have seen this on my Atlas account as well :(

**Comment by radeksimko** [comment]
Fix in #2620

**Comment by radeksimko** [comment]
Fixed via #2620
